genesyscloud/resource_genesyscloud_webdeployments_configuration.go

```diff
@@ -0,0 +1,888 @@
+				Description: "Details about a selector event trigger",
+				Type:        schema.TypeList,
+				Optional:    true,
+				Elem:        selectorEventTrigger,
+			},
+			"scroll_depth_event": {
+				Description: "Details about a scroll percentage event trigger",
+				Type:        schema.TypeList,
+				Optional:    true,
+				Elem:        scrollPercentageEventTrigger,
+			},
+		},
+	}
+)
+
+func getAllWebDeploymentConfigurations(ctx context.Context, clientConfig *platformclientv2.Configuration) (ResourceIDMetaMap, diag.Diagnostics) {
+	resources := make(ResourceIDMetaMap)
+	webDeploymentsAPI := platformclientv2.NewWebDeploymentsApiWithConfig(clientConfig)
+
+	configurations, _, getErr := webDeploymentsAPI.GetWebdeploymentsConfigurations(false)
+	if getErr != nil {
+		return nil, diag.Errorf("Failed to get web deployment configurations: %v", getErr)
+	}
+
+	for _, configuration := range *configurations.Entities {
+		resources[*configuration.Id] = &ResourceMeta{Name: *configuration.Name}
+	}
+
+	return resources, nil
+}
+
+func webDeploymentConfigurationExporter() *ResourceExporter {
+	return &ResourceExporter{
+		GetResourcesFunc: getAllWithPooledClient(getAllWebDeploymentConfigurations),
+	}
+}
+
+func resourceWebDeploymentConfiguration() *schema.Resource {
+	return &schema.Resource{
+		Description: "Genesys Cloud Web Deployment Configuration",
+
+		CreateContext: createWithPooledClient(createWebDeploymentConfiguration),
+		ReadContext:   readWithPooledClient(readWebDeploymentConfiguration),
+		UpdateContext: updateWithPooledClient(updateWebDeploymentConfiguration),
+		DeleteContext: deleteWithPooledClient(deleteWebDeploymentConfiguration),
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		SchemaVersion: 1,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description:  "Deployment name",
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 100),
+			},
+			"description": {
+				Description: "Deployment description",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"languages": {
+				Description: "A list of languages supported on the configuration.",
+				Type:        schema.TypeList,
+				Optional:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"default_language": {
+				Description: "The default language to use for the configuration.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"status": {
+				Description: "The current status of the deployment. Valid values: Pending, Active, Inactive, Error, Deleting.",
+				Type:        schema.TypeString,
+				Computed:    true,
+				Optional:    true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"Pending",
+					"Active",
+					"Inactive",
+					"Error",
+					"Deleting",
+				}, false),
+				DiffSuppressFunc: validateConfigurationStatusChange,
+			},
+			"version": {
```

Comment:
The version is usually left out of resources and the latest version is retrieved from the API using a GET call prior to updating it. You can see that pattern applied in other resources

Reply:
I believe it's necessary in this case since it needs to be referenced by the deployment, but since it's computed it functions similarly to how you describe. The difference being that updates don't automatically use the latest version. I could switch it to do that though.

Reply:
Oh yeah sorry, the other resources use versions a bit differently. Just put it as `Optional: true` and it'll be fine. I tried exporting some configurations using the current schema and then applying them and got an error because of that.